Makefile

```diff
@@ -26,11 +26,11 @@ VFATQC_VER_MINOR=4
 VFATQC_VER_PATCH=3
 
 include $(BUILD_HOME)/$(Project)/config/mfCommonDefs.mk
-include $(BUILD_HOME)/$(Project)/config/mfPythonDefs.mk
```

Comment:
why was this removed? it was only in the last commit

Reply:
I removed it because there is now no ``mfPythonDefs.mk`` in the ``config`` submodule, at least on the branch i pulled config submodules from

Reply:
The hash i pulled from being ``ea9f181``

Reply:
Hmm, according to the log and graph, this file was added in the second commit on the `gembuild` (`config`) repo and never changed (until after the commit you moved the submodule to).

Can you do one final rebase?
```
git rebase -i 8973d4d
```
and execute the following actions (the hashes here are what I see from a current clone of your repo, but they should be the three hashes that `git` is acting on when you actually do the rebase, so don't just copy/paste these)
```
pick 94d8647 First draft Sphinx Documentation
edit 63ff627 bumped config submodule to add sphinx
drop 7067185 bumped config submodule to include Sphinx support
```

During the `edit`, can you do (this will move the config submodule a bit further, adding the `pip` `make` target):
```
cd config
git checkout 8163a1f
cd -
git add config
git commit -m "bumped config submodule to add sphinx and pip targets"
git rebase --continue
```

Reply:
Thanks! Updated now, though that hash doesn't seem to have an ``mfPythonDefs.mk`` file so now make doc fails again...

Reply:
That's weird... After I've merged with your most recent branch, it was missing few files in config, notably `mfSphinx.mk`. However, after I executed the code snippet from @jsturdy 
```bash
cd config
git checkout 8163a1f
cd -
git add config
git commit -m "bumped config submodule to add sphinx and pip targets"
```
the files has been added and I was able to compile the doc successfully.
In case you're interested, my version of your code is here:
https://github.com/mexanick/vfatqc-python-scripts/tree/develop

Reply:
That worked and I have bumped that submodule as well as some ``.rst`` html template files

Reply:
One must remember that when merging/rebasing, submodules might have to be done separately
```
git submodule update --init --recursive
```
otherwise, the submodule *may* still point to whatever you had in the local repo previously

Reply:
e.g., from my active development tree:

```
[sw/vfatqc-python-scripts]% git co -b wren-rpc-playground-PR wren/rpc-playground
M       config
Branch 'wren-rpc-playground-PR' set up to track remote branch 'rpc-playground' from 'wren' by rebasing.
Switched to a new branch 'wren-rpc-playground-PR'
[sw/vfatqc-python-scripts]% git submodule update --init --recursive
Submodule path 'config': checked out '8163a1f2089ef62675036ebdd6b916365d04d6eb'
[sw/vfatqc-python-scripts]% make doc
PythonModules=["gempython.vfatqc", "gempython.vfatqc.utils", ]
Using BUILD_HOME=/home/sturdy/src/sw
OS Detected: centos7
make html
PythonModules=["gempython.vfatqc", "gempython.vfatqc.utils", ]
Using BUILD_HOME=/home/sturdy/src/sw
OS Detected: centos7
make[1]: Entering directory `/data/bigdisk/users/sturdy/software/vfatqc-python-scripts'
python /usr/bin/sphinx-build -b html -d /home/sturdy/src/sw/vfatqc-python-scripts/doc/_build/doctrees   /home/sturdy/src/sw/vfatqc-python-scripts/doc /home/sturdy/src/sw/vfatqc-python-scripts/doc/_build/html
Making output directory...
Running Sphinx v1.1.3
loading pickled environment... not yet created
loading intersphinx inventory from http://docs.python.org/objects.inv...
building [html]: targets for 14 source files that are out of date
updating environment: 14 added, 0 changed, 0 removed
Traceback (most recent call last):canV3                                                                                                                                                                                                                                                                                                                                      
  File "/usr/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 321, in import_object
    __import__(self.modname)
  File "/data/bigdisk/users/sturdy/software/vfatqc-python-scripts/dacScanV3.py", line 68, in <module>
    from gempython.tools.hw_constants import maxVfat3DACSize
ImportError: No module named gempython.tools.hw_constants
Traceback (most recent call last):Chamber                                                                                                                                                                                                                                                                                                                                    
  File "/usr/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 321, in import_object
    __import__(self.modname)
  File "/data/bigdisk/users/sturdy/software/vfatqc-python-scripts/trimChamber.py", line 94, in <module>
    from gempython.tools.vfat_user_functions_uhal import *
ImportError: No module named gempython.tools.vfat_user_functions_uhal
Traceback (most recent call last):                                                                                                                                                                                                                                                                                                                                           
  File "/usr/lib/python2.7/site-packages/sphinx/ext/autodoc.py", line 321, in import_object
    __import__(self.modname)
  File "/data/bigdisk/users/sturdy/software/vfatqc-python-scripts/run_scans.py", line 92, in <module>
    from gempython.gemplotting.mapping.chamberInfo import chamber_config
ImportError: No module named gempython.gemplotting.mapping.chamberInfo
reading sources... [100%] virtualenv-p5                                                                                                                                                                                                                                                                                                                                      
/home/sturdy/src/sw/vfatqc-python-scripts/doc/man/dacScanV3.rst:1: WARNING: autodoc can't import/find module 'dacScanV3', it reported error: "No module named gempython.tools.hw_constants", please check your spelling and sys.path
/home/sturdy/src/sw/vfatqc-python-scripts/doc/man/trimChamber.rst:1: WARNING: autodoc can't import/find module 'trimChamber', it reported error: "No module named gempython.tools.vfat_user_functions_uhal", please check your spelling and sys.path
/home/sturdy/src/sw/vfatqc-python-scripts/doc/runscans.rst:25: WARNING: autodoc can't import/find module 'run_scans', it reported error: "No module named gempython.gemplotting.mapping.chamberInfo", please check your spelling and sys.path
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] virtualenv-p5                                                                                                                                                                                                                                                                                                                                       
None:None: WARNING: toctree contains reference to document 'man/dacScanV3' that doesn't have a title: no link will be generated
None:None: WARNING: toctree contains reference to document 'man/trimChamber' that doesn't have a title: no link will be generated
/home/sturdy/src/sw/vfatqc-python-scripts/doc/runscans.rst:7: WARNING: toctree contains reference to document 'man/dacScanV3' that doesn't have a title: no link will be generated
/home/sturdy/src/sw/vfatqc-python-scripts/doc/runscans.rst:7: WARNING: toctree contains reference to document 'man/trimChamber' that doesn't have a title: no link will be generated
writing additional files... (0 module code pages) genindex py-modindex search
copying static files... WARNING: html_static_path entry '/home/sturdy/src/sw/vfatqc-python-scripts/doc/_static' does not exist
done
dumping search index... done
dumping object inventory... done
build succeeded, 8 warnings.

Build finished. The HTML pages are in /home/sturdy/src/sw/vfatqc-python-scripts/doc/_build/html.
make[1]: Leaving directory `/data/bigdisk/users/sturdy/software/vfatqc-python-scripts'
```

Reply:
I actually do not have that many warnings...
```bash
make cleandoc
PythonModules=["gempython.vfatqc", "gempython.vfatqc.utils", ]
Using BUILD_HOME=/home/mdalchen/work/cms-gem-daq-project
OS Detected: centos7
rm -rf /home/mdalchen/work/cms-gem-daq-project/vfatqc-python-scripts/doc/_build/*
-bash-4.2$ make doc
PythonModules=["gempython.vfatqc", "gempython.vfatqc.utils", ]
Using BUILD_HOME=/home/mdalchen/work/cms-gem-daq-project
OS Detected: centos7
make html
PythonModules=["gempython.vfatqc", "gempython.vfatqc.utils", ]
Using BUILD_HOME=/home/mdalchen/work/cms-gem-daq-project
OS Detected: centos7
make[1]: Entering directory `/home/mdalchen/work/cms-gem-daq-project/vfatqc-python-scripts'
python /usr/bin/sphinx-build -b html -d /home/mdalchen/work/cms-gem-daq-project/vfatqc-python-scripts/doc/_build/doctrees   /home/mdalchen/work/cms-gem-daq-project/vfatqc-python-scripts/doc /home/mdalchen/work/cms-gem-daq-project/vfatqc-python-scripts/doc/_build/html
Making output directory...
Running Sphinx v1.1.3
loading pickled environment... not yet created
loading intersphinx inventory from http://docs.python.org/objects.inv...
building [html]: targets for 14 source files that are out of date
updating environment: 14 added, 0 changed, 0 removed
ImportError: Using gempython_gemplotting package default for chamber_config dictionary
ImportError: Using gempython_gemplotting package default for GEBtype dictionary
ImportError: Using gempython_gemplotting package default for chamber_vfatDACSettings dictionary
reading sources... [100%] virtualenv-p5
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] virtualenv-p5
writing additional files... (1 module code pages) _modules/index
 genindex py-modindex search
copying static files... WARNING: html_static_path entry '/home/mdalchen/work/cms-gem-daq-project/vfatqc-python-scripts/doc/_static' does not exist
done
dumping search index... done
dumping object inventory... done
build succeeded, 1 warning.

Build finished. The HTML pages are in /home/mdalchen/work/cms-gem-daq-project/vfatqc-python-scripts/doc/_build/html.
make[1]: Leaving directory `/home/mdalchen/work/cms-gem-daq-project/vfatqc-python-scripts'
```

Reply:
I am getting the same output as Misha on a clean install :
```
$ git clone git@github.com:vetens/vfatqc-python-scripts.git
Cloning into 'vfatqc-python-scripts'...
remote: Enumerating objects: 91, done.
remote: Counting objects: 100% (91/91), done.
remote: Compressing objects: 100% (74/74), done.
remote: Total 6227 (delta 35), reused 57 (delta 17), pack-reused 6136
Receiving objects: 100% (6227/6227), 22.97 MiB | 12.12 MiB/s, done.
Resolving deltas: 100% (3611/3611), done.
[wvetens@gem904daq01]:/afs/cern.ch/work/w/wvetens/cms-gem-daq/tmp
$ cd vfatqc-python-scripts/
[wvetens@gem904daq01]:/afs/cern.ch/work/w/wvetens/cms-gem-daq/tmp/vfatqc-python-scripts
$ git co rpc-playground 
Branch rpc-playground set up to track remote branch rpc-playground from origin.
Switched to a new branch 'rpc-playground'
[wvetens@gem904daq01]:/afs/cern.ch/work/w/wvetens/cms-gem-daq/tmp/vfatqc-python-scripts
$ git submodule update --init --recursive
Submodule 'config' (https://github.com/cms-gem-daq-project/gembuild.git) registered for path 'config'
Cloning into 'config'...
remote: Enumerating objects: 87, done.
remote: Counting objects: 100% (87/87), done.
remote: Compressing objects: 100% (60/60), done.
remote: Total 352 (delta 55), reused 54 (delta 27), pack-reused 265
Receiving objects: 100% (352/352), 123.90 KiB | 0 bytes/s, done.
Resolving deltas: 100% (223/223), done.
Submodule path 'config': checked out '8163a1f2089ef62675036ebdd6b916365d04d6eb'
[wvetens@gem904daq01]:/afs/cern.ch/work/w/wvetens/cms-gem-daq/tmp/vfatqc-python-scripts
$ make doc                                                                                                                                                                                                                                                                                
PythonModules=["gempython.vfatqc", "gempython.vfatqc.utils", ]
Using BUILD_HOME=/afs/cern.ch/work/w/wvetens/cms-gem-daq/tmp
OS Detected: centos7
make html
PythonModules=["gempython.vfatqc", "gempython.vfatqc.utils", ]
Using BUILD_HOME=/afs/cern.ch/work/w/wvetens/cms-gem-daq/tmp
OS Detected: centos7
make[1]: Entering directory `/afs/cern.ch/work/w/wvetens/cms-gem-daq/tmp/vfatqc-python-scripts'
python /usr/bin/sphinx-build -b html -d /afs/cern.ch/work/w/wvetens/cms-gem-daq/tmp/vfatqc-python-scripts/doc/_build/doctrees   /afs/cern.ch/work/w/wvetens/cms-gem-daq/tmp/vfatqc-python-scripts/doc /afs/cern.ch/work/w/wvetens/cms-gem-daq/tmp/vfatqc-python-scripts/doc/_build/html
Making output directory...
Running Sphinx v1.1.3
loading pickled environment... not yet created
loading intersphinx inventory from http://docs.python.org/objects.inv...
building [html]: targets for 14 source files that are out of date
updating environment: 14 added, 0 changed, 0 removed
ImportError: Using gempython_gemplotting package default for chamber_config dictionary                                                                                                                                                                                                    
ImportError: Using gempython_gemplotting package default for GEBtype dictionary
ImportError: Using gempython_gemplotting package default for chamber_vfatDACSettings dictionary
reading sources... [100%] virtualenv-p5                                                                                                                                                                                                                                                   
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] virtualenv-p5                                                                                                                                                                                                                                                    
writing additional files... (1 module code pages) _modules/index
 genindex py-modindex search
copying static files... WARNING: html_static_path entry '/afs/cern.ch/work/w/wvetens/cms-gem-daq/tmp/vfatqc-python-scripts/doc/_static' does not exist
done
dumping search index... done
dumping object inventory... done
build succeeded, 1 warning.

Build finished. The HTML pages are in /afs/cern.ch/work/w/wvetens/cms-gem-daq/tmp/vfatqc-python-scripts/doc/_build/html.
make[1]: Leaving directory `/afs/cern.ch/work/w/wvetens/cms-gem-daq/tmp/vfatqc-python-scripts'
```

Reply:
Those errors and warnings don't seem to affect the actual html output that is produced by make doc so we may (?) be able to ignore them

Reply:
yeah, i actually didn't care about any of the warnings in my build, since it succeeded, which was the point i was making (i wasn't building from in a virtualenv or with any expectation that the dependencies would be present)
from my point of view, @vetens' code is doing what it's expected to do, so we can remove WIP and proceed with the review